go/nbs/table_reader.go

```diff
@@ -389,36 +378,32 @@ func (tr tableReader) calcReads(reqs []getRecord, blockSize, maxReadSize, ampThr
 	// into large physical reads.
 	sort.Sort(offsetRecords)
 
-	for i, rec := range offsetRecords {
-		if reqs[rec.reqIdx].data != nil {
-			continue // already satisfied
-		}
-
-		// Mark this request "satisfied"
-		reqs[rec.reqIdx].data = []byte{0x01}
+	var readStart, readEnd uint64
+	readStarted := false
 
-		readStart := rec.offset
-		readEnd := rec.offset + uint64(tr.lengths[rec.ordinal]) // implicitly include the first chunk
+	for i := 0; i < len(offsetRecords); {
+		rec := offsetRecords[i]
+		length := tr.lengths[rec.ordinal]
 
-		// As we scan forward, for each location/length, we'll include it in the current read if
-		// the total number of bytes we'll read contains fewer than X bytes we don't care about.
-		readAmp := uint64(0)
+		if !readStarted {
+			readStarted = true
```

Comment:
I think maybe that, if we increment `reads` every time we set `readStarted` to true, that's equivalent to what we're doing now, and it lets us get rid of the extra check after the loop, which feels a bit clunky.